File: Data Structures & Algorithms/three-integer-sum/submission-0.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        answer = []
        for i in range(len(nums)):
            for j in range(i+1,len(nums)):
                for k in range(j+1, len(nums)):
                    if nums[i] + nums[j] + nums[k] == 0:
                        numbers = [nums[i],nums[j],nums[k]]
                        numbers.sort()
                        if numbers not in answer:
                            answer.append(numbers)
        return answer

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        priority_queue<pair<int,int>> frequency;
        map<int,int> numbers;
        // sort count into numbers
        for(auto x : nums){
            numbers[x] += 1;
        }
        // add the value into the frequency
        for(auto x : numbers){
            frequency.push({x.second,x.first});
        }
        // pop the top most k values
        vector<int> answer;
        for(int i = 0; i < k; i++){
            answer.push_back(frequency.top().second);
            frequency.pop();
        }
        return answer;
    }
};
